### PR TITLE
👌 IMPROVE: Workflows - Work chains module

### PR DESCRIPTION
**Pull request by mbercx**
Make several small changes in language and format to the first half of the
work chain module in the "Workflows" section.

Also add some exercises after the participants have run their first work
chain.

The PR also adds some new MyST substitutions for icons used by the margin
information, and fixes the regex for the `sphinx-copybutton`.

### CODE REVIEW

**Comment by mbercx** [comment]
@ramirezfranciscof in this module we use the `run()` method to launch the toy work chains for the tutorial, since they complete almost instantly and it avoids the issue with the daemon `PYTHONPATH`. Do you show them how to `run` workflows (or calculations) anywhere? I added **TODO ADD LINK** for now, but want to link to that module section when we first do this.

File: docs/sections/workflows/workchain.md
Comment (flavianojs) [inline docs/sections/workflows/workchain.md:170]: Do you know why when we use the snippet button to copy this command, we get the command plus two blank lines? This makes the command to be executed automatically when pasted on a terminal.

**Comment by ramirezfranciscof** [comment]
> Do you show them how to `run` workflows (or calculations) anywhere?

Don't they `run` calculations/workflows in the `Fundamentals -> Basics` section? I am using submit for most of the things I show except for the dry run but even there I was kind of assuming they already would knew how to run stuff.

**Comment by mbercx** [comment]
> Don't they `run` calculations/workflows in the `Fundamentals -> Basics` section? I am using submit for most of the things I show except for the dry run but even there I was kind of assuming they already would knew how to run stuff.

True, they do. Was thinking you might also add a `run` example in the "Running computations" section somewhere, but I suppose that most would take too long to complete. Might be nice to show the issue with running here though, your call! :)

File: docs/sections/workflows/workchain.md
Comment (mbercx) [reply docs/sections/workflows/workchain.md:170]: Hmm, Indeed. Let me check the `sphinx-copybutton` regex again.

**Review by flavianojs — COMMENTED** [review]
Thanks, Marnik. All seem good. And the exercises are great addition to the material.
After checking the snippet copying button, it should be good to Merge.

File: docs/sections/workflows/workchain.md
Comment (mbercx) [reply docs/sections/workflows/workchain.md:170]: Strange, if I test this via https://regex101.com/, it seems to only give 1 match, so not sure where the added empty lines come from.

I suggest we simply fix this by splitting up the in- and output in two code-blocks for now.

**Comment by mbercx** [comment]
> Thanks, Marnik. All seem good. And the exercises are great addition to the material.
> After checking the snippet copying button, it should be good to Merge.

Great, thanks for the review @flavianojs! One subtle change that I was still wondering about is whether to "dedent" the `code-block`s, e.g. as in the Define subsection:

https://aiida-tutorials--362.org.readthedocs.build/en/362/sections/workflows/workchain.html#define-method

Or stick with the indented code snippets as e.g. in the "How not to create data" section:

https://aiida-tutorials--362.org.readthedocs.build/en/362/sections/workflows/workchain.html#how-not-to-create-data

What do you think?

**Comment by flavianojs** [comment]
> > Thanks, Marnik. All seem good. And the exercises are great addition to the material.
> > After checking the snippet copying button, it should be good to Merge.
> 
> Great, thanks for the review @flavianojs! One subtle change that I was still wondering about is whether to "dedent" the `code-block`s, e.g. as in the Define subsection:
> 
> https://aiida-tutorials--362.org.readthedocs.build/en/362/sections/workflows/workchain.html#define-method
> 
> Or stick with the indented code snippets as e.g. in the "How not to create data" section:
> 
> https://aiida-tutorials--362.org.readthedocs.build/en/362/sections/workflows/workchain.html#how-not-to-create-data
> 
> What do you think?

I had thought of the dedent myself without arriving at a conclusion. Therefore, either away work for me. The visual without the indentation is better, but the visual association with the script is better with the indentation... I guess I would leave the indentation to prioritize the functionality.

File: docs/sections/workflows/workchain.md
Comment (mbercx) [reply docs/sections/workflows/workchain.md:170]: Alright, the empty lines are added as well to the copied content. Might need to raise an issue on https://github.com/executablebooks/sphinx-copybutton, but for now we just keep this in mind for code snippets.

Comment (flavianojs) [reply docs/sections/workflows/workchain.md:170]: Alright then.

**Comment by mbercx** [comment]
@flavianojs ready for review! I've also used the margin to remind the participant that in the `verdi process show <PK>` code snippet, they have to replace the `<PK>`:

https://aiida-tutorials--362.org.readthedocs.build/en/362/sections/workflows/workchain.html#run-the-work-chain

Think it's a good idea to add this to the margin consistently?

**Review by flavianojs — APPROVED** [review]
I think you can merge it now.

**Comment by flavianojs** [comment]
> @flavianojs ready for review! I've also used the margin to remind the participant that in the `verdi process show <PK>` code snippet, they have to replace the `<PK>`:
> 
> https://aiida-tutorials--362.org.readthedocs.build/en/362/sections/workflows/workchain.html#run-the-work-chain
> 
> Think it's a good idea to add this to the margin consistently?

I don't find it essential, but it won't hurt. Reminding them to replace the PK once on the first occasion during the tutorial is essential, but more than that is accessory.

**Comment by mbercx** [comment]
> I think you can merge it now.

But I need your "approval" 😀 

Like official-GitHub-PR-approval :)

**Review by flavianojs — APPROVED** [review]
My blessing.

**Review by flavianojs — APPROVED** [review]
All good.

**Comment by mbercx** [comment]
@flavianojs Ah, my bad! I hadn't noticed you had actually approved, because it was still blocked. But the reason it wasn't allowing me to merge is because you didn't have write access yet. 😅

**Pull request merged**
